tests/unit/hive_metastore/test_workflows.py

```diff
@@ -53,6 +55,26 @@ def test_migrate_ctas_views(run_workflow):
     ctx.workspace_client.catalogs.list.assert_called()
 
 
+class MockBackendFriend(MockBackend):
```

Comment:
@nfx : I would like you opinion on this class. Alternatively, a change to lsql makes sense:
- Change `rows_written_for` as below. Note: this changes the return type(!!!)
- Introduce new public method `has_rows_written_for` that returns `bool` to indicate `found_write_match` below

Reply:
no return type changes :) introduce new method in lsql as a follow-up